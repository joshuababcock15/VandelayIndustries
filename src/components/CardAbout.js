import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { CardBody, CardTitle, CardSubtitle, Card, List } from 'reactstrap';

const propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
  imageAlt: PropTypes.string,
  imageSrc: PropTypes.string,
  listTitle: PropTypes.string,
  lists: PropTypes.array,
};

const defaultProps = {};

const CardWrapper = styled.div`
  width: 100%;
  padding: 40px;
  display: flex;
  justify-content: space-evenly;
`;
const ContentWrapper = styled.div`
  max-width: 400px;
`;

const CardAbout = (props) => {
  const { title, subtitle, imageSrc, imageAlt, listTitle, lists } = props;

  return (
    <CardWrapper>
      <Card color="light">
        <ContentWrapper>
          <CardBody>
            <CardTitle tag="h5">{title}</CardTitle>
            <CardSubtitle className="mb-2 text-muted" tag="h6">
              {subtitle}
            </CardSubtitle>
            <List type="unstyled">
              <li>
                {listTitle}
                <ul>
                  {lists &&
                    lists.map((item, index) => (
                      <li key={index}>{item?.list}</li>
                    ))}
                </ul>
              </li>
            </List>
          </CardBody>
        </ContentWrapper>
      </Card>
      <img alt={imageAlt} src={imageSrc} width="30%" />
    </CardWrapper>
  );
};

CardAbout.propTypes = propTypes;
CardAbout.defaultProps = defaultProps;

export default CardAbout;
