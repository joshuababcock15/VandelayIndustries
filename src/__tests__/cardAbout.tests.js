import React from 'react';
import renderer from 'react-test-renderer';
import CardAbout from '../components/CardImage';
import jerry from '../assets/jerry.png';

const listData = {
  lists: [
    {
      list: 'Integrity',
    },
    {
      list: 'Cheapness',
    },
    {
      list: 'Did I mention being cheap',
    },
    {
      list: 'Latex',
    },
  ],
};

describe('CardAbout', () => {
  const tree = renderer.create(<CardAbout title="Vandley Industries" />);
  const testInstance = tree.toJSON();

  test('Matches snapshot', () => {
    expect(testInstance).toMatchSnapshot();
  });

  test('default props', () => {
    const testRenderer = renderer.create(
      <CardAbout
        title="test title"
        subtitle="test subtitle"
        listTitle="list title"
        lists={listData?.lists}
        imageAlt="jerry"
        imageSrc={jerry}
      />
    );
    const testInstances = testRenderer.root;
    expect(testInstances.props.title).toBe('test title');
    expect(testInstances.props.subtitle).toBe('test subtitle');
    expect(testInstances.props.listTitle).toBe('list title');
    expect(testInstances.props.lists).toBe(listData?.lists);
    expect(testInstances.props.imageSrc).toBe(jerry);
    expect(testInstances.props.imageAlt).toBe('jerry');
  });
});
