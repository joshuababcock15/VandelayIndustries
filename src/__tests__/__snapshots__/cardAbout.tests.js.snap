// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardAbout Matches snapshot 1`] = `
<div
  className="card"
>
  <img
    className="card-img-top"
    width="100%"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Vandley Industries
    </h5>
    <p
      className="card-text"
    >
      
    </p>
    <a />
  </div>
</div>
`;

exports[`CardImage Matches snapshot 1`] = `
<div
  className="card"
>
  <img
    className="card-img-top"
    width="100%"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Vandley Industries
    </h5>
    <p
      className="card-text"
    >
      
    </p>
    <a />
  </div>
</div>
`;
